include/glow/Base/Tensor.h

```diff
@@ -48,6 +49,66 @@ void genericTranspose(const Tensor *src, Tensor *dest,
 /// returned dims. For example, input {2,1,4} would result in {2,1,4,1,1,1}.
 ShapeVector expandDimsToMax(llvm::ArrayRef<size_t> currDims);
 
+namespace runtime {
+class DeviceManager;
+}
+
+class DeviceResidencyInfo final {
+  enum class TensorResidency {
+    Host,
+    Device,
+  };
+
+  /// The residency status of the tensor.
+  TensorResidency tensorResidency_{TensorResidency::Host};
```

Comment:
Nit: You may want to move the declaration of `tensorResidency_` below `deviceManager_` and `deviceManager_` pointers. Due to the alignments rules for class/struct fields, this may reduce the size of `DeviceResidencyInfo` objects. I.e. `sizeof(DeviceResidencyInfo_with_suggested_changes)` will be very likely < `sizeof(DeviceResidencyInfo)`.